plugins/library/messagehub.py

```diff
@@ -0,0 +1,202 @@
+      - Mutually exclusive with env-variable
+    required: false
+    default: 1
+
+  env-variable:
+    description:
+      - Name of environmental variable which contains CI message in .json format.
+      - Mutually exclusive with user, password, selector, host, port, destination and count
+    required: true
+    default: None
+
+  output:
+    description:
+      - Output metadata file name where CI Message data will be stored.
+    required: false
+    default: metamorph.json
+'''
+
+EXAMPLES = '''
+- name: Get single message from message bus
+  messagehub:
+    user: "..."
+    password: "..."
+    host: "..."
+  register: result
+
+- name: Get single message from environmental variable
+  messagehub:
+    env-variable: "..."
+  register: result
+
+- name: Get single message from environmental variable and store it into hello.json
+  messagehub:
+    env-variable: "..."
+    output: "hello.json"
+  register: result
+>>>>>>> Added support of different json output names.
+'''
+
+RETURN = '''
+Error message:
+    description: Received error message from message bus
+    returned: Error
+    type: dictionary
+    sample: {"header": "...", "message": "Error description"}
+CI message:
+    description: CI messages from message bus
+    returned: changed
+    type: dictionary
+    sample: {"messages": [...]}
+'''
+
+
+import stomp
+import logging
+import logging.config
+import time
+import os
+from lib.logging_conf import setup_logging, storing_pretty_json
+from ansible.module_utils.basic import AnsibleModule
+
+
+class CIListener(stomp.ConnectionListener):
+    def __init__(self, count):
+        self.count = count
+        self.metamorph_data = []
+        self.error_message = {}
+
+    def on_error(self, headers, message):
+        self.error_message['headers'] = headers
+        self.error_message['message'] = message
+
+    def on_message(self, headers, message):
+        if self.count > len(self.metamorph_data):
+            self.metamorph_data.append({"header": headers, "message": message})
+            self.count += 1
+
+
+def messagebus_run(module):
+    conn = stomp.Connection([(module.params['host'], module.params['port'])])
+    listener = CIListener(module.params['count'])
+    conn.set_listener('CI Listener', listener)
+    conn.start()
+    conn.connect(login=module.params['user'], passcode=module.params['password'], wait=True)
+    if module.params['selector']:
+        conn.subscribe(
+            destination=module.params['destination'],
+            id='1',
+            ack='auto',
+            headers={'selector': module.params['selector']}
+        )
+    else:
+        conn.subscribe(
+            destination=module.params['destination'],
+            id='1',
+            ack='auto'
+        )
+    logging.info("Connection to message bus established.")
+    while (not listener.error_message) and len(listener.metamorph_data) < module.params['count']:
+        logging.info("Waiting 1s for CI message to arrive")
```

Comment:
Can we have this as logging.debug ? logging.info for every 1s is too much of an output.